From bc616e7f6bad1a6938259a1281087ee75aa6bd0f Mon Sep 17 00:00:00 2001
From: Patrick Harbers <jgrharbers@gmail.com>
Date: Sat, 16 Apr 2016 23:48:18 +0200
Subject: [PATCH] From: Intel/HazouPH

The Wi-Fi card doesn't power up again after going into
power-saving mode. MMC tries to open it, but can't connect.
The Card driver should resume after the sdio driver
goes out of suspend modus. Not before, causing the crash.

Change-Id: I5c9edad94fd83946c2fd06b54e90626fa5234b1b
---
 mac80211/compat_wl12xx/drivers/net/wireless/wl12xx/sdio.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/mac80211/compat_wl12xx/drivers/net/wireless/wl12xx/sdio.c b/mac80211/compat_wl12xx/drivers/net/wireless/wl12xx/sdio.c
index ddf87bf..826ee98 100644
--- a/mac80211/compat_wl12xx/drivers/net/wireless/wl12xx/sdio.c
+++ b/mac80211/compat_wl12xx/drivers/net/wireless/wl12xx/sdio.c
@@ -135,6 +135,9 @@ static int wl12xx_sdio_power_on(struct wl12xx_sdio_glue *glue)
 	struct sdio_func *func = dev_to_sdio_func(glue->dev);
 	struct mmc_card *card = func->card;
 
+	/* Avoid manual resume */
+	func->card->host->bus_resume_flags = 0;
+
 	ret = pm_runtime_get_sync(&card->dev);
 	if (ret) {
 		/*
@@ -175,6 +178,10 @@ static int wl12xx_sdio_power_off(struct wl12xx_sdio_glue *glue)
 	/* Let runtime PM know the card is powered off */
 	pm_runtime_put_sync(&card->dev);
 
+	/* Enable manual resume */
+	func->card->host->bus_resume_flags |= MMC_BUSRESUME_MANUAL_RESUME;
+
+
 out:
 	return ret;
 }
-- 
2.5.0

